.github/workflows/build-windows.yml

```diff
@@ -4,75 +4,308 @@ on:
+    if: github.repository == 'dragonmux/crunch'
+    name: 'coverage ${{ matrix.os }} (msvc ${{ matrix.arch }}, ${{ matrix.cpp_std }})'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - windows-2019
+          - windows-2022
+        arch: 
+          - x86
+          - amd64
+        cpp_std:
+          - 'c++14'
+          - 'c++17'
+      fail-fast: false
+    env:
+      CC: cl.exe
+      CXX: cl.exe
+      LD: link.exe
+    steps:
+      - name: Runtime environment
+        shell: bash
+        env:
+          WORKSPACE: ${{ github.workspace }}
+        run: |
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          echo "GITHUB_WORKSPACE=`pwd`" >> $GITHUB_ENV
+      - name: Coverage environment
+        run: |
+          echo "BUILD_OPTS=--buildtype=debug" >> $GITHUB_ENV
+      - name: Setup compiler
         uses: ilammy/msvc-dev-cmd@v1
         with:
           arch: ${{ matrix.arch }}
-      - name: Coverage environment
+      - name: Setup OpenCppCoverage
         shell: bash
-        env:
-          COVERAGE: ${{ matrix.coverage }}
-        run: |
-          echo "Coverage enabled for build? ${{ matrix.coverage }}"
-          echo -n "COVERAGE=" >> $GITHUB_ENV
-          ([ $COVERAGE == "true" ] && echo -n 1 || echo -n 0) >> $GITHUB_ENV
-          if [ $COVERAGE == "true" ]; then
-            echo "BUILD_OPTS=--buildtype=debug" >> $GITHUB_ENV
-            curl -L1O https://github.com/OpenCppCoverage/OpenCppCoverage/releases/download/release-0.9.9.0/OpenCppCoverageSetup-x64-0.9.9.0.exe
-            MSYS2_ARG_CONV_EXCL=/dir=\;/verysilent ./OpenCppCoverageSetup-x64-0.9.9.0 \
-              /dir="C:\Program Files\OpenCppCoverage" /verysilent
-            rm OpenCppCoverageSetup-x64-0.9.9.0.exe
-          else
-            echo "BUILD_OPTS=" >> $GITHUB_ENV
-          fi
-        working-directory: ${{ runner.temp }}
+        run: |
+          curl -L1O https://github.com/OpenCppCoverage/OpenCppCoverage/releases/download/release-0.9.9.0/OpenCppCoverageSetup-x64-0.9.9.0.exe
+          MSYS2_ARG_CONV_EXCL=/dir=\;/verysilent ./OpenCppCoverageSetup-x64-0.9.9.0 \
+            /dir="C:\Program Files\OpenCppCoverage" /verysilent
+          rm OpenCppCoverageSetup-x64-0.9.9.0.exe
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          lfs: true
+          submodules: true
       - name: Setup Meson + Ninja
         shell: bash
         run: |
-          curl -L1O https://github.com/ninja-build/ninja/releases/download/v1.10.0/ninja-win.zip
-          python -m pip install --upgrade pip setuptools wheel
-          pip3 install meson
-          unzip ninja-win.zip -d /c/windows/system32
-          rm ninja-win.zip
+          python3 -m pip install --upgrade pip setuptools wheel
+          python3 -m pip install meson ninja
         working-directory: ${{ runner.temp }}
       - name: Version tools
         shell: bash
         run: |
-          rm /usr/bin/link.exe
```

Comment:
Guessing this no longer ends up pointing at the wrong link.exe?

Reply:
Correct, see https://github.com/actions/runner-images/commit/7c96951971ca872b4f02469a3f945af20f517d2b